x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/DeploymentManager.java

```diff
@@ -77,51 +74,49 @@ public DeploymentManager(Client client, NamedXContentRegistry xContentRegistry,
+    private void doStartDeployment(TrainedModelDeploymentTask task, ActionListener<TrainedModelDeploymentTask> listener) {
         logger.debug("[{}] Starting model deployment", task.getModelId());
 
         ProcessContext processContext = new ProcessContext(task.getModelId(), task.getIndex(), executorServiceForProcess);
 
-        if (processContextByAllocation.putIfAbsent(task.getAllocationId(), processContext) != null) {
-            throw ExceptionsHelper.serverError("[{}] Could not create process as one already exists", task.getModelId());
+        if (processContextByAllocation.putIfAbsent(task.getId(), processContext) != null) {
+            listener.onFailure(ExceptionsHelper.serverError("[{}] Could not create process as one already exists", task.getModelId()));
+            return;
         }
 
         String taskConfigDocId = NlpTaskConfig.documentId(task.getModelId());
 
         ActionListener<Boolean> modelLoadedListener = ActionListener.wrap(
             success -> {
                 executorServiceForProcess.execute(() -> processContext.resultProcessor.process(processContext.process.get()));
-
-                setTaskStateToStarted(task, ActionListener.wrap(
-                    response -> logger.info("[{}] trained model loaded", task.getModelId()),
-                    e -> failTask(task,
-                        String.format(Locale.ROOT, "[%s] error setting task state to [%s] [%s]",
-                            task.getModelId(), TrainedModelDeploymentState.STARTED, e))
-                ));
+                listener.onResponse(task);
             },
-            e -> failTask(task,
-                String.format(Locale.ROOT, "[%s] error loading model [%s]", task.getModelId(), e))
+            listener::onFailure
         );
 
         ActionListener<SearchResponse> configListener = ActionListener.wrap(
             searchResponse -> {
                 if (searchResponse.getHits().getHits().length == 0) {
-                    failTask(task, Messages.getMessage(Messages.TASK_CONFIG_NOT_FOUND, task.getModelId(), taskConfigDocId));
+                    listener.onFailure(
+                        new ResourceNotFoundException(
+                            Messages.getMessage(Messages.TASK_CONFIG_NOT_FOUND, task.getModelId(), taskConfigDocId)
+                        )
+                    );
                     return;
                 }
 
                 NlpTaskConfig config = parseConfigDocLeniently(searchResponse.getHits().getAt(0));
                 NlpTask nlpTask = NlpTask.fromConfig(config);
                 NlpTask.Processor processor = nlpTask.createProcessor();
                 processContext.nlpTaskProcessor.set(processor);
-                startAndLoad(task, processContext, modelLoadedListener);
+                // This comes back on possibly a network thread, don't lock it up with a start
+                executorServiceForProcess.execute(() -> startAndLoad(processContext, modelLoadedListener));
```

Comment:
I am gonna pull this out of here and open a separate PR. Discovered this bug in testing, as the search response may be handled on a network thread, so any handling must be done asynchronously and I think opening the file handles could take a bit of time and blocks the thread.